internal/authentication/tokengetter.go

```diff
@@ -19,6 +21,21 @@ type TokenGetter struct {
 	mu                 sync.RWMutex
 }
 
+type ServiceAccountNotFoundError struct {
+	ServiceAccountName      string
+	ServiceAccountNamespace string
+	Err                     error
+}
+
+func (e *ServiceAccountNotFoundError) Unwrap() error {
+	return e.Err
+}
+
+// Error implements the error interface for ServiceAccountNotFoundError.
+func (e *ServiceAccountNotFoundError) Error() string {
+	return fmt.Sprintf("service account %q not found in namespace %q: unable to authenticate with the Kubernetes cluster.", e.ServiceAccountName, e.ServiceAccountNamespace)
```

Comment:
```suggestion
	return fmt.Sprintf(
		"service account \"%s\" not found in namespace \"%s\": "+
			"unable to authenticate with the Kubernetes cluster. (%s)",
		e.ServiceAccountName,
		e.ServiceAccountNamespace,
		e.Err.Error())
```

Reply:
We need also return the default error value

Reply:
ok